src/test/java/com/volasoftware/tinder/services/implementations/FriendServiceImplTest.java

```diff
@@ -258,10 +279,88 @@ public void testLinkAllAccountsWithBotsWhenRealAccountsDoesNotHaveBotsThenSucces
         assertEquals(OperationConstant.SUCCESSFUL, result);
     }
 
+
+    @Test
+    public void testSortFriendsByLocation() {
+        //given
+        FriendSearchDto friendSearchDto = new FriendSearchDto();
+        friendSearchDto.setLatitude(0.0);
+        friendSearchDto.setLongitude(0.0);
+
+        List<FriendDto> expectedFriends = Arrays.asList(
+            FriendDto.builder().firstName("John").lastName("Doe")
+                .gender(Gender.MALE).age(25)
+                .locationDto(LocationDto.builder().latitude(10.0).longitude(10.0).build()).build(),
+            FriendDto.builder().firstName("Jane").lastName("Doe")
+                .gender(Gender.FEMALE).age(22)
+                .locationDto(LocationDto.builder().latitude(20.0).longitude(20.0).build()).build(),
+            FriendDto.builder().firstName("Alice").lastName("Doe")
+                .gender(Gender.OTHER).age(30)
+                .locationDto(LocationDto.builder().latitude(30.0).longitude(30.0).build()).build()
+        );
+
+        //when
+        List<Account> friendsList = Arrays.asList(friend1, friend2, friend3);
+        ArrayList<Account> friends = new ArrayList<>(friendsList);
+        when(accountRepository.findOneByEmail(any())).thenReturn(
+            Optional.ofNullable(loggedAccount));
+        when(friendMapper.accountListToFriendDtoList(friends)).thenReturn(expectedFriends);
+
+        List<FriendDto> expectedFriendDtos = friendMapper.accountListToFriendDtoList(friends);
+        List<FriendDto> sortedFriendDtos = friendService.sortFriendsByLocation(friendSearchDto);
+
+        //then
+        assertEquals(expectedFriendDtos, sortedFriendDtos);
+    }
+
     private Account createAccount(Long id, AccountType type) {
         Account account = new Account();
         account.setId(id);
         account.setAccountType(type);
         return account;
     }
-}
\ No newline at end of file
+
+    private void createAccountWithFriends() {
+        loggedAccount = new Account();
+        loggedAccount.setId(1L);
+        loggedAccount.setEmail("user1");
+
+        friend1 = new Account();
```

Comment:
You can extract the creation of a single account in separate method and pass different data as arguments of it without repeating the whole thing over and over again.

Reply:
> I have just saw this kind of method I was talking about "createAccount()" maybe extend it with the new required data.

I renamed "createAccount()" to "createAccountByType()", because it has different purpose.